src/python/pants/core/goals/update_build_files.py

```diff
@@ -218,10 +222,21 @@ async def update_build_files(
     )
 
     rewrite_request_classes = []
+    formatter_to_request_class: dict[Formatter, type[RewrittenBuildFileRequest]] = {
+        Formatter.BLACK: FormatWithBlackRequest,
+        Formatter.YAPF: FormatWithYapfRequest,
+        Formatter.RUFF: FormatWithRuffRequest,
+    }
     for request in union_membership[RewrittenBuildFileRequest]:
-        if issubclass(request, (FormatWithBlackRequest, FormatWithYapfRequest)):
-            is_chosen_formatter = issubclass(request, FormatWithBlackRequest) ^ (
-                update_build_files_subsystem.formatter == Formatter.YAPF
+        if issubclass(
+            request, (FormatWithBlackRequest, FormatWithRuffRequest, FormatWithYapfRequest)
+        ):
+            chosen_formatter_request_class = formatter_to_request_class.get(
+                update_build_files_subsystem.formatter
```

Comment:
This looks to be independent of the loop? Potentially it could move out, and this check could become something like:

```python
chosen_formatter_request_class = formatter_to_request_class.get(update_build_files_subsystem.formatter)

for request in union_membership[RewrittenBuildFileRequest]:
    if update_build_files_subsystem.fmt and issubclass(request, chosen_formatter_request_class):
        rewrite_request_classes.append(request)
    if update_build_files_subsystem.fix_safe_deprecations or ...:
        ...
```

Reply:
Excellent point! Added this change